### ISSUE TITLE
MAC - Implement Read-Aloud (selected Text)

### ISSUE THREAD

**Issue by nachochris** [oregondesignservices/MorphicLiteClientMac#31]
We need to have a button on Mac that is Read-Selected text (like we do on Windows).  This should be the read-aloud option on the bar, rather than toggle VoiceOver on/off

**Comment by christopher-rtf** [comment]
A little bootstrapping: here is a quick chunk of code which sends the Option-Esc key sequence to the system (which should start and stop macOS's native "read text"-style feature).  Please note that this relies on some assumptions (appropriate window has focus, OS feature actually does what we want it to, keystroked are picked up by the OS 100% of the time when sent this way, etc.).  And of note: there are also multiple ways to send keys to processes or the HID stream.

` import Carbon.HIToolbox
let escapeKeyCode: CGKeyCode = CGKeyCode(kVK_Escape)
guard let combinedSessionStateEventSource: CGEventSource = CGEventSource(stateID: CGEventSourceStateID.combinedSessionState) else {
    fatalError() // TODO: degrade gracefully
}
guard let optionEscKeyDownEvent: CGEvent = CGEvent(keyboardEventSource: combinedSessionStateEventSource, virtualKey: escapeKeyCode, keyDown: true) else {
    fatalError() // TODO: degrade gracefully
}
optionEscKeyDownEvent.flags = CGEventFlags.maskAlternate
guard let optionEscKeyUpEvent: CGEvent = CGEvent(keyboardEventSource: combinedSessionStateEventSource, virtualKey: escapeKeyCode, keyDown: false) else {
    fatalError() // TODO: degrade gracefully
}

sleep(3) // for demo purposes only: give the user 3 seconds to select something before we send the option-escape keypress

optionEscKeyDownEvent.post(tap: CGEventTapLocation.cgSessionEventTap)
optionEscKeyUpEvent.post(tap: CGEventTapLocation.cgSessionEventTap)`

**Comment by christopher-rtf** [comment]
This issue has been resolved via PR #46.
